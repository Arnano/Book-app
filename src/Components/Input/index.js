import React from 'react';
import PropTypes from 'prop-types';
import StyledInput from './style';

const Input = ({ classes, onBlur, onChange, onClick, type, placeholder, value }) => 
  <StyledInput
    className={classes}
    onBlur={onBlur}
    onChange={onChange}
    onClick={onClick}
    type={type}
    placeholder={placeholder}
    value={value}
  />;

Input.defaultProps = {
  placeholder: '',
  type: 'text'
}

Input.propTypes = {
  classes: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ])
}

export default Input;
