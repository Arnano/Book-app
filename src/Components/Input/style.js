import Styled from 'styled-components';
import { blue } from '../../Style/colors';

const StyledInput = Styled.input `
  border: 1px solid ${blue};
  opacity: 0.8;
  width: 300px;
  padding: 8px 0;
  border-radius: 4px;
`;

export default StyledInput;