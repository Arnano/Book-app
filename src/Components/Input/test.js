import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input  from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Input', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Input />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Input />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
