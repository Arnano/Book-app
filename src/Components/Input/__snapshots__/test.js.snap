// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input has a valid snapshot 1`] = `
<input
  className="sc-bdVaJa mnJkf"
  onBlur={undefined}
  onChange={undefined}
  onClick={undefined}
  placeholder=""
  type="text"
  value={undefined}
/>
`;
